nncf/quantization/algorithms/channel_alignment/openvino_backend.py

```diff
@@ -0,0 +1,162 @@
+from nncf.openvino.graph.transformations.commands import OVTargetPoint
+from nncf.openvino.graph.transformations.commands import OVWeightUpdateCommand
+from nncf.openvino.statistics.collectors import OVNNCFCollectorTensorProcessor
+from nncf.openvino.statistics.collectors import OVQuantileReducer
+from nncf.openvino.statistics.statistics import OVMinMaxTensorStatistic
+from nncf.quantization.algorithms.channel_alignment.backend import ALGO_BACKENDS
+from nncf.quantization.algorithms.channel_alignment.backend import ChannelAlignmentAlgoBackend
+from nncf.quantization.algorithms.channel_alignment.backend import LayoutDescriptor
+
+
+@ALGO_BACKENDS.register(BackendType.OPENVINO)
+class OVChannelAlignmentAlgoBackend(ChannelAlignmentAlgoBackend):
+    @staticmethod
+    def target_point(target_type: TargetType, target_node_name: str, port_id: int) -> OVTargetPoint:
+        return OVTargetPoint(target_type, target_node_name, port_id)
+
+    @staticmethod
+    def get_bias_value(node: NNCFNode, model: ov.Model, nncf_graph: NNCFGraph) -> np.ndarray:
+        return get_bias_value(node, nncf_graph, model)
+
+    @staticmethod
+    def get_weight_value(node: NNCFNode, model: ov.Model, port_id: int) -> np.ndarray:
+        return get_weight_value(node, model, port_id)
+
+    @staticmethod
+    def get_activation_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        return 0, 0
+
+    @staticmethod
+    def get_weights_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        return 0, 1
+
+    @staticmethod
+    def get_conv_metatypes():
+        return [OVConvolutionMetatype, OVGroupConvolutionMetatype, OVDepthwiseConvolutionMetatype]
+
+    @staticmethod
+    def get_linear_metatypes():
+        return [OVMatMulMetatype]
+
+    @staticmethod
+    def get_add_metatypes():
+        return [OVAddMetatype, OVSubtractMetatype]
+
+    @staticmethod
+    def get_statistic_collector(
+        reduction_shape, q: float, num_samples: int, inplace: bool
+    ) -> TensorStatisticCollectorBase:
+        tensor_collector = TensorCollector(OVMinMaxTensorStatistic)
+        quantile_reducer = OVQuantileReducer(reduction_shape, (q, 1 - q), inplace)
+
+        for port_id, container_key in enumerate([OVMinMaxTensorStatistic.MIN_STAT, OVMinMaxTensorStatistic.MAX_STAT]):
+            aggregator = MedianAggregator(OVNNCFCollectorTensorProcessor, num_samples=num_samples)
+            tensor_collector.register_statistic_branch(container_key, quantile_reducer, aggregator, port_id)
+        return tensor_collector
+
+    @staticmethod
+    def is_node_with_bias(node: NNCFNode, nncf_graph: NNCFGraph) -> bool:
+        next_nodes = nncf_graph.get_next_nodes(node)
+        if not next_nodes:
+            return False
+
+        add_node = next_nodes[0]
+        if add_node.metatype != OVAddMetatype:
+            return False
+
+        bias_constant = get_node_with_bias_value(add_node, nncf_graph)
+        return bias_constant is not None
```

Comment:
Could you please explain why you don't use an existing implementation?

https://github.com/openvinotoolkit/nncf/blob/a1bec7147a675a6bf76b5a386c8660e36d4aba71/nncf/openvino/graph/node_utils.py#L31

Reply:
That's because `is_node_with_bias` is using following check:
https://github.com/openvinotoolkit/nncf/blob/263ec98825c1d639c61e635b755df0d1dace2f3d/nncf/openvino/graph/node_utils.py#L38-L39

And this list is used in BC and FBC algorithms
https://github.com/openvinotoolkit/nncf/blob/develop/nncf/openvino/graph/metatypes/openvino_metatypes.py#L668-L672

I need this check to work on group convolutions and I don't want to alter BC and FBC behavior, so I decided to skip this check. Besides this check is not needed as `is_node_with_bias` inside CA algorithm is always called on convolutional or linear nodes